app/assets/onepager/js/send.js

```diff
@@ -168,12 +288,41 @@ function validateEmail(email) {
   return re.test(email);
 }
 
+function validateStreamingToken() {
+  var tokenAddress = (
+    ($('#token').val() == '0x0') ?
+      '0x0000000000000000000000000000000000000000'
+      : $('#token').val());
+  // derived info
+  var isSendingETH = (tokenAddress == '0x0' || tokenAddress == '0x0000000000000000000000000000000000000000');
+  var tokenDetails = tokenAddressToDetails(tokenAddress);
+  // TODO: Extra validation: whitelisted tokens for streaming?
+
+  return !isSendingETH;
+}
+
 function isNumeric(n) {
   return !isNaN(parseFloat(n)) && isFinite(n);
 }
 
+function getEffectiveStreamDeposit(amount, decimals, time) {
+  var amountBN = web3.toBigNumber('1'.padEnd(decimals + 1, 0)).mul(web3.toBigNumber(amount * 1.0));
+
+  return amountBN.sub(amountBN.mod(web3.toBigNumber(time)));
+}
+
+function transactionReceiptAsync(txHash, f) {
+  var transactionReceipt = web3.eth.getTransactionReceipt(txHash, function(error, receipt) {
+    if (receipt) {
+      f(receipt);
+    } else {
+      setTimeout(() => transactionReceiptAsync(txHash, f), 1000);
+    }
+  });
+}
+
 
-function sendTip(email, github_url, from_name, username, amount, comments_public, comments_priv, from_email, accept_tos, tokenAddress, expires, success_callback, failure_callback, is_for_bounty_fulfiller) {
+function sendTip(email, github_url, from_name, username, amount, streamTime, comments_public, comments_priv, from_email, accept_tos, tokenAddress, expires, success_callback, failure_callback, is_for_bounty_fulfiller) {
```

Comment:
there are other areas in the codebase that `sendTip()` is called. ull need to migrate them all.

Reply:
Thanks for pointing that out 😄